earth_enterprise/src/common/geFilePool.cpp

```diff
@@ -20,17 +20,65 @@
 #include <third_party/rsa_md5/crc32.h>
 #include <khEndian.h>
 
+// ****************************************************************************
+// ***  POSIXFileAcessor
+// ****************************************************************************
+int POSIXFileAccessor::Open(const std::string &fname, int flags, mode_t createMask) {
+  return khOpen(fname, flags, createMask);
+}
+
+int POSIXFileAccessor::FsyncAndClose(AbstractFileIdentifier* pID) {
+  return khFsyncAndClose(pID->getAsFD());
+}
+
+int POSIXFileAccessor::Close(AbstractFileIdentifier* pID) {
+  return khClose(pID->getAsFD());
+}
+
+bool POSIXFileAccessor::PreadAll(AbstractFileIdentifier* pID, void* buffer, size_t size, off64_t offset) {
+  return khPreadAll(pID->getAsFD(), buffer, size, offset);
+}
+
+bool POSIXFileAccessor::PwriteAll(AbstractFileIdentifier* pID, const void* buffer, size_t size, off64_t offset) {
+  return khPwriteAll(pID->getAsFD(), buffer, size, offset);
+}
+
+// ****************************************************************************
+// ***  FileIdentifierFactory
+// ****************************************************************************
+namespace FileIdentifierFactory {
+  static AbstractFileIdentifier* getIdentifier(int i) {
```

Comment:
Probably best to have this return `shared_ptr<AbstractFileIdentifier>` instead of `AbstractFileIdentifier*`. Then return `std::make_shared<POSIXIdentifier>(i)`.